hooks/path_info.py

```diff
@@ -266,11 +266,24 @@ def get_frame_sequence_path(self, path, frame_spec=None):
 
         path_template = self.sgtk.template_from_path(path)
         if path_template:
+            # TODO: copied from get_path_for_frame() - refactor this.
+            seq_key = None
             # if the path fits a template, use that and check if it is a file sequence
-            if "SEQ" in path_template.keys:
-                fields = path_template.get_fields(path)
-                del fields["SEQ"]
-                return path_template.apply_fields(fields)
+            for key in path_template.keys.values():
+                if isinstance(key, SequenceKey):
+                    seq_key = key
+                    break
+
+            if seq_key:
+                if seq_key.name in path_template.keys:
```

Comment:
Isn't this conditional redundant? You got the seq_key from the template's keys list already, so you shouldn't need to check the existence again.

Reply:
Good point.